hack/get-terraform.sh

```diff
@@ -1,7 +1,15 @@
 #!/bin/sh
 
+OS=linux
+ARCH=amd64
+if command -v go >/dev/null 2>/dev/null
+then
+	OS="$(go env GOOS)" &&
+	ARCH="$(go env GOARCH)"
+fi &&
 TERRAFORM_VERSION="0.11.8" &&
-TERRAFORM_URL="https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip" &&
+TERRAFORM_URL="https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_${OS}_${ARCH}.zip" &&
+echo "pulling ${TERRAFORM_URL}" >&2 &&
 cd "$(dirname "$0")/.." &&
 mkdir -p bin &&
 curl -L "${TERRAFORM_URL}" | funzip > bin/terraform &&
```

Comment:
gunzip?

Reply:
> s/funzip/gunzip ?

```console
$ wget https://releases.hashicorp.com/terraform/0.11.8/terraform_0.11.8_linux_amd64.zip
$ file terraform_0.11.8_linux_amd64.zip 
terraform_0.11.8_linux_amd64.zip: Zip archive data, at least v2.0 to extract
```

`funzip` is about extracting files from ZIP archives (what we want here).  `gunzip` is about removing gzip compression.  It looks like they both function equivalently for this single-file ZIP, but `funzip` is a more accurate semantic match (I can't find docs for `gunzip` being able to handle ZIP archives, what happens if Hashicorp adds another file to the zip?).

Did you have specific reasons for changing this?

Reply:
I was running through the installer and didn't have funzip installed and couldn't find it in the fedora repos.  Your reasoning for using funzip makes sense but maybe we need a note saying it is a dependency? (unless I missed it somewhere).  It was easy enough to workaround it.

Reply:
> I was running through the installer and didn't have funzip installed...

I've pushed a belt-and-suspenders 78bf9f7dc ;).

> ... and couldn't find it in the fedora repos...

I expect it's in `unzip` based on [this old `Dockerfile`][1].

[1]: https://github.com/openshift/installer/commit/64e02b3be1cae3d311feb48f41152a7b1f8bf305#diff-e93aaab1fdfc92bac48708be1c4df739L35